test/integration/userStatus.test.js

```diff
@@ -103,21 +103,165 @@ describe("UserStatus", function () {
     });
   });
 
+  describe("PATCH /users/status/update", function () {
+    let testUserId;
+    let testUserJwt;
+    let clock;
+    beforeEach(async function () {
+      clock = sinon.useFakeTimers({
+        now: new Date(2022, 10, 12).getTime(),
+        toFake: ["Date"],
+      });
+      testUserId = await addUser(userData[1]);
+      testUserJwt = authService.generateAuthToken({ userId: testUserId });
+    });
+
+    afterEach(function () {
+      clock.restore();
+    });
+
+    it("Should update the User Status based on the future dates", async function () {
+      // creating Active Status from 12th Nov 2022 (1669401000000)
+      const response1 = await chai
+        .request(app)
+        .patch(`/users/status/self`)
+        .set("Cookie", `${cookieName}=${testUserJwt}`)
+        .send(generateUserStatusData("ACTIVE", 1669401000000, 1669401000000));
+      expect(response1).to.have.status(201);
+      expect(response1.body.data.currentStatus.state).to.equal("ACTIVE");
+
+      // Marking OOO Status from 24th Nov 2022 to 28th Nov 2022
+      const response2 = await chai
+        .request(app)
+        .patch(`/users/status/self`)
+        .set("Cookie", `${cookieName}=${testUserJwt}`)
+        .send(generateUserStatusData("OOO", 1668215609000, 1669228200000, 1669573800000, "Vacation Trip"));
```

Comment:
doesn't the test seem too brittle?

Reply:
can you please explain more. what do you mean when you say brittle? what should I add more.